datatypes/src/raster/geo_transform.rs

```diff
@@ -83,13 +94,20 @@ impl GeoTransform {
     /// use geoengine_datatypes::primitives::{Coordinate2D};
     ///
     /// let geo_transform = GeoTransform::new_with_coordinate_x_y(0.0, 1.0, 0.0, -1.0);
-    /// assert_eq!(geo_transform.coordinate_2d_to_grid_2d((0.0, 0.0).into()), (0, 0))
+    /// assert_eq!(geo_transform.coordinate_2d_to_grid_2d((0.0, 0.0).into()), [0, 0].into())
     /// ```
     ///
-    pub fn coordinate_2d_to_grid_2d(&self, coord: Coordinate2D) -> (usize, usize) {
-        let grid_x_index = ((coord.x - self.upper_left_coordinate.x) / self.x_pixel_size) as usize;
-        let grid_y_index = ((coord.y - self.upper_left_coordinate.y) / self.y_pixel_size) as usize;
-        (grid_y_index, grid_x_index)
+    pub fn coordinate_2d_to_grid_2d(&self, coord: Coordinate2D) -> GridIdx2D {
```

Comment:
put `index` or `idx` into the name, as the output is not a grid.
also: can you make this function generic over the dimension? the name is quite long

Reply:
k